src/base/bittorrent/session.cpp

```diff
@@ -2204,8 +2204,8 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
     // processed or adding to session
     if (m_addingTorrents.contains(hash) || m_loadedMetadata.contains(hash)) return false;
 
-    if (m_torrents.contains(hash)) {
-        TorrentHandle *const torrent = m_torrents.value(hash);
+    TorrentHandle *const torrent = m_torrents.value(hash);
+    if (torrent) {
```

Comment:
`if (TorrentHandle *const torrent = m_torrents.value(hash))` would be even better

Reply:
Pardon, I particularly dislike this style, I wish to stay as is, at least before C++17.

Reply:
OK